runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -620,15 +620,20 @@ class INTERPRETER_CLASS
 			UDATA result = 0;
 			do {
 				preCount = (UDATA)_sendMethod->extra;
-				postCount = preCount - _currentThread->jitCountDelta;
-				if (J9_ARE_NO_BITS_SET(preCount, J9_STARTPC_NOT_TRANSLATED)) {
-					/* Already compiled */
+				if (startAddressIsCompiled(preCount)) {
+					if (immediatelyRunCompiledMethod) {
+						if (methodCanBeRunCompiled(_sendMethod)) {
+							rc = promotedMethodOnTransitionFromJIT(REGISTER_ARGS, (void*)_pc, (void*)preCount);
+							goto done;
+						}
+					}
 					break;
```

Comment:
We can compress the code by removing one of the `if` statements.
```suggestion
				if (immediatelyRunCompiledMethod && startAddressIsCompiled(preCount)) {
					if (methodCanBeRunCompiled(_sendMethod)) {
						rc = promotedMethodOnTransitionFromJIT(REGISTER_ARGS, (void*)_pc, (void*)preCount);
						goto done;
					}
					break;
```

Reply:
No, that code is incorrect - the compiled check must occur first.